src/main/java/com/limechain/rpc/methods/state/StateRPCImpl.java

```diff
@@ -104,14 +107,18 @@ public List<String> stateGetKeysPaged(final String prefixHex, int limit, String
      * @param blockHashHex The block hash in hexadecimal format to scope the query.
      * @return The storage value in hexadecimal format for the given key within the specified block, or null if not found.
      */
+    // TODO: Figure out whether incoming requests will only ask for state in finalised block (i.e. db persisted blocks)
+    //  or also unfinalised blocks must be supported (i.e. these in BlockState.BlockTree)
+    //  For now we only handle persisted blocks (because we only read from the DB through trieStorage)
     public String stateGetStorage(final String keyHex, final String blockHashHex) {
```

Comment:
That's a mismatch in the implementation of this method: it fetches state root from `BlockState`, which also responds with unfinalised blocks, but then we query the `trieStorage`, which only handles persisted (finalised) blocks.

That's only an issue if an RPC request comes for an unfinalised block; so if that's never gonna happen, it's a no-issue (but still weird when inspecting due to this mismatch).

Reply:
Well currently we don't have the state for unfinalized block whatsoever. Maybe we may want to start executing blocks from blockannounce and keeping their states until a finalization and pruning occures, but I doubt anyone will be sending rpc requests for unfinalized blocks